#! /bin/sh /usr/share/dpatch/dpatch-run
## 30_2.6.28_support.dpatch by Mario Limonciello <superm1@ubuntu.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Pull changeset 19245 into -fixes. Fixes FTBFS on kernel 2.6.28

@DPATCH@
diff -urNad mythtv-0.21.0+fixes19136~/libs/libmythtv/dvbtypes.h mythtv-0.21.0+fixes19136/libs/libmythtv/dvbtypes.h
--- mythtv-0.21.0+fixes19136~/libs/libmythtv/dvbtypes.h	2007-01-07 09:29:30.000000000 -0600
+++ mythtv-0.21.0+fixes19136/libs/libmythtv/dvbtypes.h	2008-12-13 21:31:56.000000000 -0600
@@ -19,7 +19,7 @@
 #include <linux/dvb/frontend.h>
 #include <linux/dvb/dmx.h>
 
-#if (DVB_API_VERSION != 3)
+#if (DVB_API_VERSION != 3 && DVB_API_VERSION != 5)
 #    error "DVB driver includes with API version 3 not found!"
 #endif
 
@@ -27,7 +27,7 @@
 #    define DVB_API_VERSION_MINOR 0
 #endif
 
-#if (DVB_API_VERSION >= 3 && DVB_API_VERSION_MINOR >= 1)
+#if ((DVB_API_VERSION == 3 && DVB_API_VERSION_MINOR >= 1) || (DVB_API_VERSION > 3))
 #    define USE_ATSC
 #else
 #warning DVB API version < 3.1
